### PR TITLE
Add support for Redis ZUNION command [WIP]

### PR DESCRIPTION
**Pull request by patrox**

### CODE REVIEW

**Comment by patrox** [comment]
@machi1990 Could you please have a look if you will have a second? I just wanted to confirm if this is an acceptable solution for commands which are not (yet) available via Vert.x Redis Client.

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientImpl.java [hunk 1]
```diff
@@ -1017,6 +1021,17 @@ public Response zscore(String arg0, String arg1) {
...
     }
 
+    @Override
+    public Response zunion(List<String> args) {
+        final io.vertx.mutiny.redis.client.Command ZUNION = Command.create("zunion", -3, 0, 0, 0, false, true, true, false);
+        final io.vertx.mutiny.redis.client.Request requestWithArgs = args.stream().reduce(
+                io.vertx.mutiny.redis.client.Request.cmd(ZUNION),
+                (request, s) -> request.arg(s),
+                (request, request2) -> request);
+
```
Comment (machi1990) [inline extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientImpl.java:1031]: Yes this is the valid approach. 

The only trick part to explain can be the creation of new command. Is there some Vertx links we can add in the documentation?

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientImpl.java [hunk 2]
```diff
@@ -5,15 +5,19 @@
...
+import io.vertx.redis.client.Request;
 import io.vertx.redis.client.Response;
 
 class RedisClientImpl implements RedisClient {
     private final RedisAPI redisAPI;
+    private final MutinyRedis mutinyRedis;
     private final Duration timeout;
 
-    public RedisClientImpl(RedisAPI redisAPI, Duration timeout) {
+    public RedisClientImpl(RedisAPI redisAPI, MutinyRedis mutinyRedis, Duration timeout) {
```
Comment (machi1990) [inline extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientImpl.java:18]: `io.quarkus.redis.client.runtime.MutinyRedis` this API is in runtime package, I am wondering if in the documentation we can use the `io.vertx.mutiny.redis.client.Redis` which IIRC can be injected based on  https://github.com/quarkusio/quarkus/blob/1064752e4e19f17edaddd0e5d73902f83aa36308/extensions/redis-client/deployment/src/main/java/io/quarkus/redis/client/deployment/RedisClientProcessor.java#L141

File: extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientImpl.java (hunk 1 above)
Comment (patrox) [reply extensions/redis-client/runtime/src/main/java/io/quarkus/redis/client/runtime/RedisClientImpl.java:1031]: @machi1990 For this purpose I would describe the procedure based on running an node app (available in `vertx-redis-client`) to generate the helper methods, as described by @pmlopes here: https://github.com/vert-x3/vertx-redis-client/issues/299#issuecomment-873386056.

So I think this is covered :)

**Comment by gsmet** [comment]
It ended up being added to the Vert.x Redis client so this PR is not useful anymore.

Thanks for your efforts though and thanks for raising awareness on this being required.

**Pull request closed without merging**